File: FindAWriter/faw.py
# Find a Writer solution in Python 2 for CodeEval.com by Steven A. Dunn

import sys

for line in open(sys.argv[1], 'r'):
  line = line.rstrip('\n')

  if not line: continue
  
  line = line.split('|')
  name = line[0]
  key = map(int, line[1].lstrip(' ').split(' '))
 
  for k in key:
    sys.stdout.write(name[k - 1])
  print

File: FindAWriter/faw.rb
# find a writer soln in rb for codeeval.com by steven a dunn

File.open(ARGV[0], 'r').each_line do |line|
    line = line.chomp

    if line.empty?
        next
    end

    line = line.split('|')
    name = line[0]
    key = line[1].chomp.split(' ').map(&:to_i)

    key.each do |k|
       print name[k - 1]
    end
    puts
end